safeweb/http.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package safeweb provides a simple, safe, and opinionated wrapper around
+// http.Server that applies some web application security defenses by default.
+// It is intended to be used in place of http.Server for internal applications
+// that reside on the Tailscale network.
+//
+// safeweb requires that applications be built using two distinct
+// http.ServeMuxes for serving browser and API clients to serve the appropriate
+// security defenses in their respective contexts. When serving requests it
+// will first attempt to route requests using the browser mux, and if no
+// matching route is found it will instead defer to the API mux.
```

Comment:
Perhaps front-load the punch line a bit?

```go
// To use safeweb, the application must separate "browser" routes from "API"
// routes, with each on its own http.ServeMux. When serving requests, the
// server will first check the browser mux, and if no matching route is found
// it will defer to the API mux.
```